src/cargo/ops/cargo_rustc/build_plan.rs

```diff
@@ -0,0 +1,105 @@
+//! A graph-like structure used to represent the rustc commands to build the project and the
+//! interdependencies between them.
+//!
+//! The BuildPlan structure is used to store the dependency graph of a dry run so that it can be
+//! shared with an external build system. Each Module in the BuildPlan comprises a single
+//! subprocess and defines the build environment, the outputs produced by the subprocess, and the
+//! dependencies on other Modules.
+
+use std::collections::{BTreeMap};
+
+use super::{Context, Unit};
+use super::context::OutputFile;
+use util::{internal, CargoResult, ProcessBuilder};
+use std::sync::Arc;
+use std::path::PathBuf;
+use serde_json;
+
+#[derive(Debug, Serialize)]
+struct Module {
+    deps: Vec<String>,
+    outputs: Vec<PathBuf>,
+    links: BTreeMap<PathBuf, PathBuf>,
+    program: String,
+    args: Vec<String>,
+    env: BTreeMap<String, String>,
+    cwd: String,
```

Comment:
I think `PathBuf` implements `Serialize`, so perhaps that could be used here instead of `String` to avoid some unwraps and `to_str`?

Reply:
Looks like PathBuf works fine for cwd- thanks!